test/e2e_federation/service.go

```diff
@@ -102,15 +103,15 @@ var _ = framework.KubeDescribe("Federated Services [Feature:Federation]", func()
 
 				if service != nil {
 					By(fmt.Sprintf("Deleting service shards and their provider resources in underlying clusters for service %q in namespace %q", service.Name, nsName))
-					cleanupServiceShardsAndProviderResources(nsName, service, clusters)
+					cleanupServiceShardsAndProviderResources(nsName, service.Name, clusters)
```

Comment:
we could always get `service.Name` for `service` right? why this change is required. Please try to keep your changes to minimal unless there is a strong case in doing so.